delta-api/src/main/java/io/cdap/delta/api/DMLEvent.java

```diff
@@ -78,6 +78,11 @@ public String getRowId() {
     return rowId;
   }
 
+  public int getSizeInBytes() {
+    // the size of all other fields are fixed, only return the dynamic size of row and previousRow
+    return operation.getSizeInBytes() + computeSizeInBytes(previousRow);
```

Comment:
would be good to have a private variable that is set lazily. If null, compute this number and set it. Otherwise, return the value. That way we prevent recomputing on each call

Reply:
good point:) done :)